### DISCUSSION TITLE
Any information about performance

### DISCUSSION THREAD

**Discussion by alexander-koval** [miguelgrinberg/python-socketio#1165] [General]
Hi. It's hard to find out, how python-socketio shows itself in comparison with original Node.js version. I didn't find any information in documentation about it. I understand that it's hard  to do a performance tests and how to decide what kind of parameters need to benchmark. But for example:

- How much connections in the same time python-socketio can process in comparison with Node.js. 
- What a performance difference in comparison with Server and AsyncServer versions? 
- Is it real to process for example 10000 connections per one process?

Best Regards.

**Comment by miguelgrinberg** [comment]
Does the JavaScript Socket.IO project publish any information about performance? I don't think they do.

It is really difficult, because there are many unknowns, such as how much memory you consume per client, how often messages are exchanged between the server and each client, what CPU(s) you have, how much RAM, your network speed, etc. You have to evaluate the performance of this package considering the particular characteristics of your hosting platform and your application.

If you want a high-level answer, I can tell you with certainty that this package runs slower than the JavaScript version, simply because Python runs slower than JavaScript. Beyond that I cannot really tell you much.

> How much connections in the same time python-socketio can process in comparison with Node.js.

This package does not have a limit on how many connections can be active at the same time. The limit comes from your operating system (max number of file handles) and your platform (available RAM, network speed, etc.).

> What a performance difference in comparison with Server and AsyncServer versions?

Regular Python and Async Python run at the same speed.

> Is it real to process for example 10000 connections per one process?

If you allocate enough resources on your server, then yes.

**Reply by v3ss0n** [reply]
are there any benchmarks?
